src/markdown.nim

```diff
@@ -81,6 +81,27 @@ from htmlgen import nil, p, br, em, strong, a, img, code, del, blockquote,
 
 from markdownpkg/entities import htmlEntityToUtf8
 
+var precompiledExp {.threadvar.}: Table[string, re.Regex]
```

Comment:
The `{.threadvar.}` annotation makes this caching thread-safe (each different thread will have to compile regular expressions by itself, but I think that this is fine)